tests/unit/query_test.py

```
@@ -245,4 +248,127 @@ def test_query_payment_requirement_defaults_to_true(query_requires_payment):
     query = Query()
     assert query._is_payment_required() == True
     # Verify that payment-requiring query also defaults to requiring payment
-    assert query_requires_payment._is_payment_required() == True
\ No newline at end of file
+    assert query_requires_payment._is_payment_required() == True
```

Ruff (0.14.10)

Avoid equality comparisons to `True`; use `query_requires_payment._is_payment_required():` for truth checks

Replace with `query_requires_payment._is_payment_required()`

(E712)